scripts/cicd/verify-gather-logs-operations.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python
+
+'''
+Validates command to gather logs from a cluster
+
+Developer usage:
+
+   ssh -i log_gathering_key use-tower2.ops.rhcloud.com -- -c <clusterName> -u <kerberosID>
+
+clusterName is checked against a list of known/valid cluster names.
+
+kerberosID is the Kerberos ID of the developer requesting logs, for tracking purposes
+'''
+
+# pylint: disable=invalid-name
+# pylint: disable=logging-not-lazy
+# pylint: disable=broad-except
+
+# The log_gathering_key is kept in the shared-secrets repo. It is generated and
+# rotated weekly by the Jenkins job cluster/rotate-log-access-key. The public
+# key in .ssh/authorized_keys looks like this:
+#
+#  command="verify-gather-logs-operations.py" ssh-rsa ThePublicKey logs_access_key_week_22-YMD-H:M:S
+
+import argparse
+import os
+import re
+import sys
+import socket
+import logging
+import logging.handlers
+
+# The list of valid cluster names to request logs from
+VALID_CLUSTER_NAMES = [
+    'free-int',
+    'free-stg',
+    'starter-us-east-1',
+    'starter-us-east-2',
+    'starter-us-west-2'
+]
+
+# The command that is invoked to perform the actual log collection.  This
+# command should expect one argument, the cluster name on which to operate, and
+# produce the logs as a tarball in stdout, which gets passed directly as the
+# output of this script.
+LOG_GATHER_CMD = '/home/opsmedic/aos-cd/git/aos-cd-jobs/tower-scripts/bin/gather-logs.sh'
+
+HOSTNAME = socket.gethostname()
+
+logger = logging.getLogger('verify_command_logger')
+logger.setLevel(logging.INFO)
+logger.addHandler(logging.handlers.SysLogHandler('/dev/log'))
+
+def valid_krbid(username):
+    ''' Sanity check that the username looks like a valid kerberos id '''
+    if re.match(r'^[a-z]+$', username) and len(username) <= 8:
```

Comment:
*OUR* kerb usernames are restricted to 8 chars, but this will break in other environments.  The username cannot be all numeric, but can contain numbers.  Doing some poking about, it seems it's ok to contain underscores as well, and can be up to 31 or 32 chars long.

Consider: ^([A-Za-z_][A-Za-z0-9_]{0,30})$